pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -903,6 +911,63 @@ func (r *OnlineUpgradeReconciler) waitForReplicateToReplicaGroupB(ctx context.Co
 	return ctrl.Result{}, r.updateOnlineUpgradeStepAnnotation(ctx, r.getNextStep())
 }
 
+func (r *OnlineUpgradeReconciler) copyRedirectStateToReplicaGroupB(ctx context.Context) (ctrl.Result, error) {
+	// Skip if we've already redirected connections
+	if vmeta.GetOnlineUpgradeStepInx(r.VDB.Annotations) > waitForConnectionRedirectInx {
+		return ctrl.Result{}, nil
+	}
+
+	mainPFacts := r.PFacts[vapi.MainCluster]
+	sbPFacts := r.PFacts[r.sandboxName]
+	mainInitiator, mainOK := mainPFacts.findFirstUpPod(false /*not allow read-only*/, "" /*arbitrary subcluster*/)
+	sbInitiator, sbOK := sbPFacts.findFirstUpPod(false /*not allow read-only*/, "" /*arbitrary subcluster*/)
+	if !mainOK || !sbOK {
+		r.Log.Info("No Up nodes found; requeueing reconciliation")
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	sbSelectCmd := []string{"-tA", "-R", ",", "-c",
```

Comment:
Can the queries be wrapped into a vcluster API call? This can be a follow-up Jira. We don't want to directly execute vsql inside the pods.

Reply:
as i said (here)[https://github.com/vertica/vertica-kubernetes/pull/909#discussion_r1737166930], this is all a workaround. it would make more sense to do the server work necessary to avoid this altogether than implement a vcluster api